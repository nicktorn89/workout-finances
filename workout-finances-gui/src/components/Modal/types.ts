import { ReactNode } from 'react';

export interface ModalProps {
  title?: string;
  isActive: boolean;
  onCancel: (e: React.MouseEvent) => void;
  onOk?: (e: React.MouseEvent) => void;
  children?: ReactNode;
}
