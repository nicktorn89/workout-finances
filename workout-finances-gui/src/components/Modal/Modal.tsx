import { default as ModalComponent } from '@material-ui/core/Modal';
import React, { memo } from 'react';
import { ModalProps } from './types';
import { 
  ModalContainer, ModalHeader, ModalContent,
  ModalFooter, OkButton, CancelButton,
} from './styled';

const Modal: React.FC<ModalProps> = ({ isActive, onCancel, onOk, children, title }) => {
  return (
    <ModalComponent
      aria-labelledby='modal-title'
      aria-describedby='modal-description'
      className='modal-window'
      open={isActive}
    >
      <ModalContainer>
        <ModalHeader variant='h6' id='modal-title'>
          {title && title}
        </ModalHeader>
        <ModalContent>
          {children}
        </ModalContent>
        <ModalFooter>
          { onOk &&
            <OkButton 
              color='primary'
              variant='contained'
              className='modal-ok'
              onClick={onOk}
            >
              Ок
            </OkButton>
          }
          
          <CancelButton 
            color='secondary'
            variant='contained'
            className='modal-cancel'
            onClick={onCancel} 
          >
           Отмена
          </CancelButton>
        </ModalFooter>
      </ModalContainer>
    </ModalComponent>
  );
};

export default memo(Modal);
