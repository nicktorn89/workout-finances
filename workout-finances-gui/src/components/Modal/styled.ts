import styled from 'styled-components';
import { Typography, Button } from '@material-ui/core';

// @ts-nocheck
export const ModalContainer = styled.div`
  width: 40rem;
  height: 30rem;
  padding: 1rem;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  background-color: white;
  z-index: 1;
  border-radius: 5px;
`;

export const ModalHeader: any = styled(Typography)`
  font-size: 14px !important;
  height: 10%;
  border-bottom: 1px solid grey;
`;

export const ModalContent = styled.div`
  display: flex;
  justify-content: center;
  height: 70%;
`;

export const ModalFooter = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  height: 20%;
`;

export const OkButton: any = styled(Button)`
  margin-right: 1rem !important;
`;

export const CancelButton: any = styled(Button)`
`;
