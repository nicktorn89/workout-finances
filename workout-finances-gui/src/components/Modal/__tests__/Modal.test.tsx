import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Modal from '../Modal';

describe('<Modal />', () => {
  const mockMouseEvent = (e: React.MouseEvent) => {};

  const ModalComponent = shallow(<Modal isActive={true} onCancel={mockMouseEvent} />);
  const HiddenModalComponent = shallow(<Modal isActive={false} onCancel={mockMouseEvent} />);
  const ModalComponentToJson = shallowToJson(ModalComponent);

  it('Should render', () => {
    expect(ModalComponent).toExist();
  });

  it('Should render ModalComponent', () => {
    expect(ModalComponent).toContainExactlyOneMatchingElement('.modal-window');
  });

  it('Shouldnt render ModalComponent when isActive = false', () => {
    expect(HiddenModalComponent.contains('.modal-window')).toBe(false);
  });
  
  it('Should be equal to snapshot', () => {
    expect(ModalComponentToJson).toMatchSnapshot();
  });
});
