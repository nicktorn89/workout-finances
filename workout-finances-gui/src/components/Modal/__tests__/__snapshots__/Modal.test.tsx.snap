// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should be equal to snapshot 1`] = `
<WithStyles(Modal)
  aria-describedby="modal-description"
  aria-labelledby="modal-title"
  className="modal-window"
  open={true}
>
  <styled.div>
    <Styled(WithStyles(Typography))
      id="modal-title"
      variant="h6"
    />
    <styled.div />
    <styled.div>
      <Styled(WithStyles(Button))
        className="modal-cancel"
        color="secondary"
        onClick={[Function]}
        variant="contained"
      >
        Отмена
      </Styled(WithStyles(Button))>
    </styled.div>
  </styled.div>
</WithStyles(Modal)>
`;
